federation/pkg/federation-controller/service/dns.go

```diff
@@ -44,6 +44,10 @@ func (s *ServiceController) getHealthyEndpoints(clusterName string, cachedServic
 			return nil, nil, nil, err
 		}
 		for _, ingress := range lbStatus.Ingress {
+			_, ok := cachedService.endpointMap[lbClusterName]
```

Comment:
No, I think lets fix it now.  The comment on cachedService.endpointMap says that 0 indicates that no endpoints exist, so lets keep true to that.  It's an easy enough fix to do now, rather than difficult debugging to do later.